### PR TITLE
Bm global slack

### PR DESCRIPTION
**Pull request by bamandel**
Also changed the way our global descriptors are shown where the description is always displayed

### CODE REVIEW

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/ChannelDescriptor.java [hunk 1]
```diff
@@ -22,21 +22,22 @@
...
 package com.synopsys.integration.alert.common.descriptor;
 
+import com.synopsys.integration.alert.common.descriptor.config.ui.ChannelDistributionUIConfig;
 import com.synopsys.integration.alert.common.descriptor.config.ui.UIConfig;
 import com.synopsys.integration.alert.common.enumeration.DescriptorType;
 
 public abstract class ChannelDescriptor extends Descriptor {
 
-    public ChannelDescriptor(final String name) {
+    private ChannelDescriptor(final String name) {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/ChannelDescriptor.java]: What is this constructor used for?

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/ChannelDescriptor.java [hunk 2]
```diff
@@ -22,21 +22,22 @@
...
         super(name, DescriptorType.CHANNEL);
     }
 
-    public ChannelDescriptor(final String name, final UIConfig distributionUIConfig) {
+    private ChannelDescriptor(final String name, final ChannelDistributionUIConfig distributionUIConfig) {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/ChannelDescriptor.java]: Having all of these private constructors feels unnecessary.

File: src/main/resources/db/changelog/alert/5.1.0/changelog-global-slack.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+    <!-- Add Global Slack permissions -->
+    <changeSet author="bmandel" id="1565635286572-1">
+        <sql dbms="h2" stripComments="true">
+            CALL ASSIGN_ROLE_PERMISSION_ALL_ACCESS('ALERT_ADMIN', 'channel_slack', 'global');
+            CALL ASSIGN_ROLE_PERMISSION_ACCESS('ALERT_JOB_MANAGER', 'channel_slack', 'global', 'READ');
+            CALL ASSIGN_ROLE_PERMISSION_ACCESS('ALERT_JOB_MANAGER', 'channel_slack', 'global', 'EXECUTE');
```
Comment (gkillough) [inline src/main/resources/db/changelog/alert/5.1.0/changelog-global-slack.xml]: Do Job Managers get any access to global normally?

Comment (bamandel) [reply src/main/resources/db/changelog/alert/5.1.0/changelog-global-slack.xml]: Job Managers in the previous version had these two global permissions. I believe it's to see what channels are available. They can't actually modify anything.

File: src/main/java/com/synopsys/integration/alert/channel/slack/descriptor/SlackDescriptor.java [hunk 4]
```diff
@@ -36,12 +36,13 @@
 
     public static final String SLACK_LABEL = "Slack";
     public static final String SLACK_URL = "slack";
+    public static final String SLACK_DESCRIPTION = "This page allows you to configure the Slack for Alert.";
```
Comment (psantos1113) [inline src/main/java/com/synopsys/integration/alert/channel/slack/descriptor/SlackDescriptor.java]: There's nothing to configure on the Global Configuration UI for Slack.  The descriptions for the channels probably need to needs to be different going forward.  Maybe:
"This channel is responsible for distributing messages to Slack."  
For example email might be:
"This channel is responsible for distributing messages to an Email SMTP server."

File: src/main/resources/db/changelog/alert/5.1.0/changelog-global-slack.xml (hunk 3 above)
Comment (psantos1113) [reply src/main/resources/db/changelog/alert/5.1.0/changelog-global-slack.xml]: Yes the Job Manager can read and test channels, but not modify the configuration.

**Pull request merged**